Source/CarthageKit/LocalRepository.swift

```diff
@@ -0,0 +1,238 @@
+// swiftlint:disable vertical_parameter_alignment_on_call
+import Foundation
+import Tentacle
+import Result
+import ReactiveSwift
+
+/**
+Local repository for storing/loading dependencies and their versions. This is for testing without requiring live connection to real repositories.
+*/
+public final class LocalRepository {
+	private let directoryURL: URL
+
+	public init(directoryURL: URL) {
+		self.directoryURL = directoryURL
+	}
+
+	public func loadPinnedVersions(for dependency: Dependency, gitReference: String? = nil) throws -> [PinnedVersion] {
+		let fileURL = try self.pinnedVersionsURL(for: dependency, gitReference: gitReference)
+		let data = try Data(contentsOf: fileURL)
+		return try JSONDecoder().decode([PinnedVersion].self, from: data)
+	}
+
+	public func loadTransitiveDependencies(for dependency: Dependency, version: PinnedVersion) throws -> [(Dependency, VersionSpecifier)] {
+		let fileURL = try self.transitiveDependenciesURL(for: dependency, version: version)
+		let data = try Data(contentsOf: fileURL)
+		let versionSpecs = try JSONDecoder().decode([DependencyVersionSpecification].self, from: data)
+		return versionSpecs.map { ($0.dependency, $0.versionSpecifier) }
+	}
+
+	private func pinnedVersionsURL(for dependency: Dependency, gitReference: String? = nil, createDirs: Bool = false) throws -> URL {
```

Comment:
I find this method confusing. 

The method is really just constructing a path for a JSON file but I don't find the name suggests that. 
Maybe naming it `canonicalPinnedVersionFileURL` would be a better choice (also possibly in an extension of Dependency).

Also it has 2 responsibilities. I would prefer it not to create any path, which is the only reason why this method throws.